app/models/user.rb

```diff
@@ -3,8 +3,12 @@ class User < ApplicationRecord
   has_many :videos, through: :user_videos
 
   validates :email, uniqueness: true, presence: true
-  validates :password, presence: true
+  validates :password, presence: true, on: :update, allow_blank: true
   validates :first_name, presence: true
   enum role: { default: 0, admin: 1 }
   has_secure_password
+
+  def bookmarked_videos
+    videos.order(:tutorial_id, :position)
+  end
```

Comment:
whoa, look at that AR call

Reply:
![](https://media0.giphy.com/media/xT5LMT7ItDSEJuKAYo/giphy.gif?cid=5a38a5a24fe148449118a94908d9abaea1ee36da14631059&rid=giphy.gif)